modules/apps/fragment/fragment-renderer-react-impl/src/main/java/com/liferay/fragment/renderer/react/internal/model/listener/FragmentEntryLinkIdentifiableOSGiService.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.fragment.renderer.react.internal.model.listener;
+
+import com.liferay.portal.kernel.module.framework.service.IdentifiableOSGiService;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(service = IdentifiableOSGiService.class)
+public class FragmentEntryLinkIdentifiableOSGiService
```

Comment:
@MrJohn1911 This will not work on cluster.

```
	private static void _onNotify(
		MethodType methodType, String osgiServiceIdentifier,
		FragmentEntryLink oldFragmentEntryLink,
		FragmentEntryLink newFragmentEntryLink) {

		FragmentEntryLinkModelListener fragmentEntryLinkModelListener =
			(FragmentEntryLinkModelListener)
				IdentifiableOSGiServiceUtil.getIdentifiableOSGiService(
					osgiServiceIdentifier);

		fragmentEntryLinkModelListener._updateNPMRegistry(
			methodType, oldFragmentEntryLink, newFragmentEntryLink);
	}
```

As you can see, the method `_onNotify` expects to obtain the `FragmentEntryLinkModelListener` from the util. However, with your change, the identifiable osgi service found would be the `FragmentEntryLinkIdentifiableOSGiService`. Thus, a class cast exception would happen.

`IdentifiableOSGiService` is used in multiple places. It basically provides a simplified way to quickly get the OSGi service instance, usually used in cluster on another node. We need to find another way to do this.

In some cases we directly make the method being called `static`, like `CompanyFeatureFlagsProvider`. But it may not fit this case, as we need to call several other osgi services.

We still have some other ways to get the real osgi service, e.g. using a service tracker or using bundle context to get the service reference. Please try to find a universal way for this, so we can remove `IdentifiableOSGiService` later.

Reply:
cc @rafaprax @tinatian

Reply:
Ok, I will take a look at that, for now I'll just remove this commit from this pull and resend. Thanks.